src/mesh/grids/non_periodic_cube.cpp

```diff
@@ -0,0 +1,66 @@
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <Sacado.hpp>
+#include "non_periodic_cube.h"
+
+namespace PHiLiP {
+namespace Grids {
+
+template<int dim, typename TriangulationType>
+void non_periodic_cube_flow(
+    TriangulationType& grid,
+    double domain_left,
+    double domain_right,
+    bool colorize,
+    bool shock_tube,
+    bool shu_osher) 
+{
+    dealii::GridGenerator::hyper_cube(grid, domain_left, domain_right, colorize);
+
+    if (shock_tube || shu_osher) {
+        for (auto cell = grid.begin_active(); cell != grid.end(); ++cell) {
+            // Set a dummy boundary ID
+            cell->set_material_id(9002);
+            if (cell == grid.begin_active())
+            {
+                for (unsigned int face = 0; face < dealii::GeometryInfo<dim>::faces_per_cell; ++face) {
+                    if (shu_osher) {
```

Comment:
instead of passing the two bools that are flow case specific; you could have `non_periodic_cube_shu_osher`; which both call `non_periodic_cube_flow` where `bool shock_tube, bool shu_osher` are replaced by `const unsigned int boundary_id` (default to `1001`) and `non_periodic_cube_shu_osher` would return `non_periodic_cube(..., 1004)`; then the switch between `non_periodic_cube` and `non_periodic_cube_shu_osher` would be handled inside the `FlowSolverCase` type object; that way we separate the logic

Reply:
Modified the function such that the boundary_id of the left face is passed in as opposed to passing in the two case specific bools.